docker.go

```diff
@@ -1593,12 +1597,14 @@ func (p *DockerProvider) GetNetwork(ctx context.Context, req NetworkRequest) (ne
 }
 
 func (p *DockerProvider) GetGatewayIP(ctx context.Context) (string, error) {
-	// Use a default network as defined in the DockerProvider
 	defaultNetwork, err := p.ensureDefaultNetwork(ctx)
 	if err != nil {
 		return "", fmt.Errorf("ensure default network: %w", err)
 	}
+	return p.getGatewayIP(ctx, defaultNetwork)
+}
 
+func (p *DockerProvider) getGatewayIP(ctx context.Context, defaultNetwork string) (string, error) {
```

Comment:
By making defaultNetwork a parameter, I was able to move the decision about whether or not to lock to the caller. This function doesn't need to decide anything about locking any more.

Reply:
I should mention that default network will soon be removed in by the context rework that's currently in progress.